safegraph_py_functions/safegraph_py_functions.py

```diff
@@ -225,3 +248,31 @@ def read_pattern_multi(path_to_pattern, compression='gzip', *args, **kwargs):
 def merge_core_pattern(core_df, patterns_df, how='inner', *args, **kwargs):
     merged_df = pd.merge(core_df, patterns_df, on='safegraph_place_id', how=how, *args, **kwargs)
     return merged_df
+
+
+### --------------------------------------- END CORE, GEO, AND PATTERNS SECTION -----------------------------------------------
+
+### --------------------------------------- SOCIAL DISTANCING SECTION -----------------------------------------------
+
+## start_range and end_range = string formated as "year-month-day" 
+    ## ex: start_range = "2020-06-01", end_range = "2020-06-07"
+
+def merge_socialDist_by_dates(path_to_social_dist,start_date,end_date, *args, **kwargs):
+    path = os.path.join(path_to_social_dist,start_date[:4])
+    if(start_date[5:7] == end_date[5:7]): # same month
+        path = os.path.join(path,end_date[5:7])
+        files = [file.path for day in os.scandir(path) for file in os.scandir(day.path)][int(start_date[-2:])-1:int(end_date[-2:])]
+    else:
+        last_day = monthrange(int(start_date[:4]), int(start_date[5:7]))[1] # get last day of month
```

Comment:
@bpblakely don't want to make you bend over backwards, but what if instead of using `monthrange` you just use something like `os.listdir()` or `os.scandir()` as you do below to empirically identify the last day sitting inside that year/month/ directory? 
Something like 
```
start_path = os.path.join(os.path.join(path, start_date[0:4]),  start_date[5:7])
start_month_last_day = max([ f for f in os.listdir(start_path)))

```

You would need to add some additional string parsing and integer conversion on the path probably, but that's the idea.

Reply:
Yeah that works, thanks for the idea

Reply:
In my newest push I just grab the last date by getting the last date in os.listdir(path) and casting that to int.
```int(os.listdir(path)[-1])```

The only flaw might be if os.listdir is ever out of order for some reason. I couldn't find a case where it was out of order for the safegraph data, but maybe it's a good idea to not make that assumption

Reply:
@bpblakely I think we should probably do a "sorted" like this for safety

`sorted(os.listdir(path))[-1]`